enhancements/hypershift/kubevirt-static-ip-address-management.md

```diff
@@ -0,0 +1,531 @@
+* As a cluster administrator, I need to be able to configure static IPs for my tenant workers, since DHCP server is not available in my network.
+
+### Goals
+
+- Enable static IP address assignment for HyperShift guest cluster nodes running on KubeVirt
+- Provide a flexible API for defining IP address pools with support for various IP range formats (CIDR, hyphenated ranges, single IPs)
+- Support both IPv4 and IPv6 addressing
+- Supporting dynamic IP pool expansion or runtime IP reallocation
+- Support KubeVirt VMs live migration so the assigned network configuration stick to the VMs after live migrating them.
+- Recover pools from cluster api provider kubevirt pod restarts or node restarts.
+
+### Non-Goals
+
+- Replacing existing network configuration mechanisms in standalone OpenShift clusters
+- Implementing a full-featured IPAM solution with conflict detection across multiple management clusters
+- Providing network configuration for components outside of HyperShift guest cluster nodes
+- Support complex network configurations like bonding of interfaces and others usually configured with ignition
+
+## Proposal
+
+This proposal introduces static IPAM capabilities through modifications to three key components:
+
+1. **Cluster API Provider KubeVirt**: Extend the API to support IP pool definitions and network configuration, implement IP allocation logic from defined pools
+2. **HyperShift**: Add network configuration options to the KubeVirt platform specification at nodepool, implement translation from this network configuration to capk ip pool and openstack network config
+3. **CoreOS Afterburn**: Enable parsing and application of OpenStack network config standard data as dracut network kernel args from cloud-init config drive for KubeVirt VMs, this is similar what is done at proxmoxve provider.
+
+### Workflow Description
+
+**cluster administrator** is a human user responsible for deploying a HyperShift hosted cluster.
+
+1. Disable the default pod network and select the network to replace it:
+   - Sets `AttachDefaultNetwork=false` in the NodePool specification
+   - Specifies an additional network attachment that will replace the default network
+   - Ensures the static IP configuration matches the custom network's name
+
+2. The cluster administrator defines an IP pool configuration in the HostedCluster nodepool specification, including:
+   - Network name (to assign a ip pool)
+   - IP address ranges in CIDR notation, hyphenated format, or as individual IPs
+   - Gateway address
+   - DNS nameservers
+
+3. The cluster administrator creates the HostedCluster resource with the KubeVirt platform type and network configuration
+
+4. HyperShift reconciles the kubevirt nodepool and create KubevirtMachineTemplate resource for the guest cluster nodes.
+   - Configure interface `eno101`, gateways, and nameservers in the OpenStack network config data (without addresses) and store at KubevirtMachineTemplate `networkData` field
```

Comment:
why this particular interface ? can we add some info on the requirements for this interface ? (I assume the user chooses an interface). I'm thinking we should be clear in which requirements are there to choose said interface.